src/webServiceClient.ts

```diff
@@ -0,0 +1,138 @@
+import http = require('http');
+import https = require('https');
+import Transaction from './request/transaction';
+import * as models from './response/models';
+import { WebServiceClientError } from './types';
+
+interface ResponseError {
+  code?: string;
+  error?: string;
+}
+
+type servicePath = 'factors' | 'insights' | 'score';
+
+export default class WebServiceClient {
+  private accountID: string;
+  private host: string;
+  private licenseKey: string;
+  private timeout: number;
+
+  public constructor(accountID: string, licenseKey: string, timeout = 3000) {
+    this.accountID = accountID;
+    this.licenseKey = licenseKey;
+    this.timeout = timeout;
+    this.host = 'minfraud.maxmind.com';
+  }
+
+  public score(transaction: Transaction): Promise<models.Score> {
+    return this.responseFor<models.Score>(
+      'score',
+      transaction.toString(),
+      models.Score
+    );
+  }
+
+  private responseFor<T>(
+    path: servicePath,
+    postData: string,
+    modelClass: any
```

Comment:
`models.Score` (and eventually `models.Score | models.Insights | models.Factors`)

Reply:
I think in this case it would be `typeof models.Score` given the arg is a class reference, but it then goes on to complain that `Argument of type 'string' is not assignable to parameter of type 'ScoreResponse'` since `data` is first a string, and then converted into an object. I don't think i'm fully grasping how this can be done without `any` in this case, so I will concede defeat.